### PR TITLE
refactor: improve error handling and traceability

### PR DESCRIPTION
**Pull request by altaywtf**

### CODE REVIEW

**Comment by altaywtf** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #39** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/39?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #36** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/36?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/39?utm_source=stack-comment).

File: lib/services/account.ts [hunk 1]
```diff
@@ -8,7 +10,7 @@ const getUserId = async () => {
   const userQuery = await supabase.auth.getUser();
 
   if (userQuery.error) {
-    throw new Error(userQuery.error.message);
+    throw userQuery.error;
```
Comment (altaywtf) [inline lib/services/account.ts:13]: this is `AuthError`

**Review by okanisildar — APPROVED** [review]
this looks amazing! 😍

**Comment by altaywtf** [comment]
DANKE

**Pull request merged**